src/utils/typed.ts

```diff
@@ -0,0 +1,3 @@
+export const isUndefined = (value: unknown): value is undefined => {
```

Comment:
I've been meaning to do a refactoring where a similar file would be introduced so the basic `typeof` wrapper functions could be aggregated from `utils/num` and `utils/shortString`. Seems like this is a convenient PR to include that change.

Reply:
yeap, something that I also had in my mind, that's why this one is still in draft and I did not updated the description because of the potential TODO's :)

Reply:
@penovicp, what's the process for introducing new versioned docs? I'm asking since we'll need to update the versioned docs since utilities are changed now

Reply:
The active pre-release documentation is available as `Next` in the dropdown, the `guides` are written manually and tracked by version control and the `API` segment is generated automatically. The versioned docs are essentially snapshots of the `Next` documentation when a starknet.js version is promoted to a full release (merged to `main`), usually there is no need change them. So the changes in this PR will be automatically available as `Next` once it is merged to develop.

And just to provide the full context, the customary approach for generating new versioned docs is for one of the maintainers to run the `[Manual] Documentation Version PR` workflow on the core repo that generates the docs and opens a corresponding PR. This can be tested on any GitHub fork or the workflow script steps can be reproduced to check the behaviour locally.